pkgs/by-name/fe/fex/package.nix

```diff
@@ -42,25 +69,66 @@ llvmPackages.stdenv.mkDerivation (finalAttrs: rec {
         libclang
       ]
     ))
-  ];
+  ] ++ lib.optionals finalAttrs.doCheck [ nasm ];
 
-  buildInputs = with qt5; [
-    qtbase
-    qtdeclarative
-    qtquickcontrols
-    qtquickcontrols2
-  ];
+  buildInputs =
+    [
+      xxHash
+      fmt
+    ]
+    ++ (with qt5; [
+      qtbase
+      qtdeclarative
+      qtquickcontrols
+      qtquickcontrols2
+    ]);
 
   cmakeFlags = [
     "-DCMAKE_BUILD_TYPE=Release"
     "-DUSE_LINKER=lld"
     "-DENABLE_LTO=True"
     "-DENABLE_ASSERTIONS=False"
+    (lib.cmakeFeature "OVERRIDE_VERSION" finalAttrs.version)
     (lib.cmakeBool "BUILD_TESTS" finalAttrs.finalPackage.doCheck)
   ];
 
   strictDeps = true;
-  doCheck = false; # broken on Apple silicon computers
+
+  # Unsupported on non-4K page size kernels (e.g. Apple Silicon)
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
```

Comment:
Unfortunately there is no way to check the kernel page size here, so people wishing to compile this on Apple Silicon will have to use `.override { doCheck = false; }`.

I think this is the right thing to do, having Hydra run the tests is more valuable than being able to build this package without changes on AS. (Most users won't be impacted since they will just pull the package from the cache anyway. And if you are developing the package on AS, a temporary change to disable the tests should not be that big of a deal. I think this is preferable to requiring developers from every other platform to manually enable the tests instead.)

Reply:
```suggestion
  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;

  nativeCheckInputs = [ nasm ];
```